File: Chapter 2/Section 2.1/holstein/OLD/holstein.cpp
/*
 * ID: ten.to.1
 * TASK: holstein
 * LANG: C++
*/

#include <iostream>
#include <fstream>

int main()
{
	std::ifstream file_in("holstein.in");
	std::ofstream file_out("holstein.out");
	
	int V; file_in >> V;
	int min_vit[V];
	for(int i = 0; i < V; i++)
	{
		int r; file_in >> r;
		min_vit[i] = r;
	}
	
	int G; file_in >> G;
	int feeds[G][V];
	for(int i = 0; i < G; i++)
	{
		for(int j = 0; j < V; j++)
		{
			int r; file_in >> r;
			feeds[i][j] = r;
		}
	}
}

File: Chapter 2/Section 2.1/holstein/OLD/holstein.py
#!/usr/bin/python

"""
ID: ten.to.1
TASK: holstein
LANG: PYTHON3
"""

from itertools import combinations
import sys

file_in  = open("holstein.in", "r")
file_out = open("holstein.out", "w")

V       = int(file_in.readline())
min_vit = list(map(int, file_in.readline().split()))

G     = int(file_in.readline())
feeds = []

for n in range(0, G):
	feed = list(map(int, file_in.readline().split()))
	feeds.append(feed)

acc = []
for n in range(0, V):
	acc.append(n)


def solve():
	for n in range(1, V + 1):
		curr = list(combinations(acc, n))
		
		for j in curr:
			test = [0] * V
			for k in j:
				for l in range(0, V):
					test[l] += (feeds[k][l])
			
			found = True
			for p in range(0, V):
				if(test[p] <= min_vit[p]):
					found = False
			
			if(found):
				return j

ans = list(solve())
file_out.write(str(len(ans)) + " ")
for n in range(0, len(ans)):
	file_out.write(str(ans[n] + 1) + " ")


